File: shared/src/main/java/ru/privatenull/pnauth/api/AdmissionDecision.java
package ru.privatenull.pnauth.api;

public record AdmissionDecision(boolean allowed, boolean forceOnlineMode, Reason reason) {
    public enum Reason {
        ALLOWED,
        ONLINE_IP_LIMIT,
        REGISTERED_IP_LIMIT,
        BANNED,
        DATABASE_ERROR,
        POLICY_DENIED
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/api/AdmissionDecision.kt
package ru.privatenull.pnauth.api

data class AdmissionDecision(
    val allowed: Boolean,
    val forceOnlineMode: Boolean,
    val reason: Reason
) {
    enum class Reason {
        ALLOWED,
        ONLINE_IP_LIMIT,
        REGISTERED_IP_LIMIT,
        BANNED,
        DATABASE_ERROR,
        POLICY_DENIED
    }
}
